google-cloud-storage/src/main/java/com/google/cloud/storage/transfermanager/DownloadResult.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.transfermanager;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.StorageException;
+import com.google.common.base.MoreObjects;
+import java.nio.file.Path;
+import java.util.Objects;
+import org.checkerframework.checker.nullness.qual.MonotonicNonNull;
+import org.checkerframework.checker.nullness.qual.NonNull;
+
+public class DownloadResult {
+
+  @NonNull private final BlobInfo input;
+  @MonotonicNonNull private final Path outputDestination;
+  @NonNull private final TransferStatus status;
+  @MonotonicNonNull private final StorageException exception;
+
+  private DownloadResult(
+      @NonNull BlobInfo input,
+      Path outputDestination,
+      @NonNull TransferStatus status,
+      StorageException exception) {
+    this.input = input;
+    this.outputDestination = outputDestination;
+    this.status = status;
+    this.exception = exception;
+  }
+
+  public @NonNull BlobInfo getInput() {
+    return input;
+  }
+
+  public @NonNull Path getOutputDestination() {
+    if (status == TransferStatus.SUCCESS) {
+      return outputDestination;
+    }
+    throw new IllegalStateException("getOutputDestination() is only valid when status is SUCCESS");
+  }
+
+  public @NonNull TransferStatus getStatus() {
+    return status;
+  }
+
+  public @NonNull StorageException getException() {
+    if (status == TransferStatus.FAILED_TO_START || status == TransferStatus.FAILED_TO_FINISH) {
```

Comment:
Yes that is expected. SKIPPED is not a valid status for DownloadResult since there is no precondition for SkipIfExists is for Uploads. Canceled would be the status if the customer chose to abort the job so there really was no exception or destination.